packages/react-components/src/components/chart/hooks/useChartsLegend.tsx

```diff
@@ -4,28 +4,68 @@ import { TableProps } from '@cloudscape-design/components/table/interfaces';
 import { GraphicComponentTextOption } from 'echarts/types/src/component/graphic/GraphicModel';
 import { SeriesOption } from 'echarts';
 import { LineSeriesOption } from 'echarts/types/src/chart/line/LineSeries';
+import { DataStream } from '@iot-app-kit/core';
+import { useHover } from 'react-use';
+
+import { borderRadiusDropdown, colorBackgroundDropdownItemHover, spaceScaledS } from '@cloudscape-design/design-tokens';
+
+import { useChartStore } from '../store';
+import { isDataStreamInList } from '../../../utils/isDataStreamInList';
 
 const useChartsLegend = ({
+  datastreams,
   graphic,
   series,
 }: {
+  datastreams: DataStream[];
   graphic: InternalGraphicComponentGroupOption[];
   series: SeriesOption[];
 }) => {
   const legendColumnDefinition = {
     id: 'Legends',
     header: <div className='base-chart-legend-col-header'>Properties</div>,
-    cell: (e: { lineColor: string; name: string }) => (
-      <div className='base-chart-legend-row-data-container'>
+    cell: (e: { datastream: DataStream; lineColor: string; name: string }) => {
+      const { datastream, lineColor, name } = e;
+      const highlightDataStream = useChartStore((state) => state.highlightDataStream);
+      const unHighlightDataStream = useChartStore((state) => state.unHighlightDataStream);
+      const highlightedDataStreams = useChartStore((state) => state.highlightedDataStreams);
+      const isDataStreamHighlighted = isDataStreamInList(highlightedDataStreams);
+
+      const toggleHighlighted = () => {
+        if (isDataStreamHighlighted(datastream)) {
+          unHighlightDataStream(datastream);
+        } else {
+          highlightDataStream(datastream);
```

Comment:
the input is ```datastream: DataStream``` so its always defined, in that case , the ignature for ```highlightDataStream``` can be ```highlightDataStream(datastream" DataStream);``` right ? i.e we can avoid if else is zustand state

Reply:
this isn't the only place we are using highlight / unhighlight. Other locations have an undefined.